Incompatible with systems using openssl 1.1

... like Debian.

I'm not sure what's up with the openssl mess in Rust. Can you poke the dependencies somehow, maybe?

Here's the failure btw:
```
error: failed to compile `rink v0.4.0`, intermediate artifacts can be found at `/tmp/cargo-install.9kFLTAXS6YhJ`

Caused by:
  failed to run custom build command for `openssl v0.7.14`
process didn't exit successfully: `/tmp/cargo-install.9kFLTAXS6YhJ/release/build/openssl-b1e8fa07f9be8a07/build-script-build` (exit code: 101)
--- stdout
TARGET = Some("x86_64-unknown-linux-gnu")
OPT_LEVEL = Some("3")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CC_x86_64-unknown-linux-gnu = None
CC_x86_64_unknown_linux_gnu = None
HOST_CC = None
CC = None
HOST = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CFLAGS_x86_64-unknown-linux-gnu = None
CFLAGS_x86_64_unknown_linux_gnu = None
HOST_CFLAGS = None
CFLAGS = None
PROFILE = Some("release")
running: "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-o" "/tmp/cargo-install.9kFLTAXS6YhJ/release/build/openssl-87c538093dbe2342/out/src/c_helpers.o" "-c" "src/c_helpers.c"
cargo:warning=src/c_helpers.c:1:25: fatal error: openssl/ssl.h: No such file or directory
cargo:warning= #include <openssl/ssl.h>
cargo:warning=                         ^
cargo:warning=compilation terminated.
ExitStatus(ExitStatus(256))
```

Ah no, it *did* break in some other way:
```
running: "cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-o" "/tmp/cargo-install.Ed1goGzG6yE3/release/build/openssl-sys-extras-45159e42f4e1f90d/out/src/openssl_shim.o" "-c" "src/openssl_shim.c"
cargo:warning=src/openssl_shim.c: In function ‘DH_new_from_params’:
cargo:warning=src/openssl_shim.c:132:7: error: dereferencing pointer to incomplete type ‘DH {aka struct dh_st}’
cargo:warning=     dh->p = p;
cargo:warning=       ^~
cargo:warning=src/openssl_shim.c: In function ‘X509_get_extensions_shim’:
cargo:warning=src/openssl_shim.c:143:13: error: dereferencing pointer to incomplete type ‘X509 {aka struct x509_st}’
cargo:warning=     return x->cert_info ? x->cert_info->extensions : NULL;
cargo:warning=             ^~
ExitStatus(ExitStatus(256))
```

--- a/.gitignore
+++ b/.gitignore
@@ -1,0 +1,3 @@
+/Cargo.lock
+/target/
+/web/rink-web.toml

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,7 +23,7 @@ strsim = "0.5.1"
 chrono-tz = "0.2.2"
 chrono-humanize = { version = "0.0.6", optional = true }
 linefeed = { version = "0.1.4", optional = true }
-hyper = { version = "0.9.10", optional = true }
+hyper = { version = "0.10.4", optional = true }
 libc = { version = "0.2.14", optional = true }
 ipc-channel = { version = "0.5.1", optional = true }
 xml-rs = { version = "0.3.4", optional = true }

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -13,7 +13,7 @@ keywords = ["unit", "math", "conversion", "cli", "tool"]
 default = ["linefeed", "chrono-humanize", "gpl", "currency"]
 sandbox = ["libc", "ipc-channel"]
 gpl = []
-currency = ["hyper", "xml-rs", "json"]
+currency = ["hyper", "hyper-native-tls", "xml-rs", "json"]
 nightly = ["serde", "serde_derive"]
 
 [dependencies]
@@ -24,6 +24,7 @@ chrono-tz = "0.2.2"
 chrono-humanize = { version = "0.0.6", optional = true }
 linefeed = { version = "0.1.4", optional = true }
 hyper = { version = "0.10.4", optional = true }
+hyper-native-tls = { version = "0.2.2", optional = true }
 libc = { version = "0.2.14", optional = true }
 ipc-channel = { version = "0.5.1", optional = true }
 xml-rs = { version = "0.3.4", optional = true }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -45,6 +45,8 @@ extern crate ipc_channel;
 #[cfg(feature = "currency")]
 extern crate hyper;
 #[cfg(feature = "currency")]
+extern crate hyper_native_tls;
+#[cfg(feature = "currency")]
 extern crate xml;
 #[cfg(feature = "currency")]
 extern crate json;
@@ -370,6 +372,8 @@ fn cached(file: &str, url: &str, expiration: Duration) -> Result<File, String> {
     use std::io::{Read, Write};
     use hyper::Client;
     use hyper::status::StatusCode;
+    use hyper::net::HttpsConnector;
+    use hyper_native_tls::NativeTlsClient;
 
     fn ts<T:Display>(x: T) -> String {
         format!("{}", x)
@@ -398,7 +402,9 @@ fn cached(file: &str, url: &str, expiration: Duration) -> Result<File, String> {
             try!(fs::create_dir_all(path.parent().unwrap()).map_err(|x| format!("{}", x)));
             let mut f = try!(File::create(tmppath.clone()).map_err(|x| format!("{}", x)));
 
-            let client = Client::new();
+            let ssl = NativeTlsClient::new().unwrap();
+            let connector = HttpsConnector::new(ssl);
+            let client = Client::with_connector(connector);
             let mut res = try!(client.get(url).send().map_err(|x| format!("{}", x)));
             if res.status != StatusCode::Ok {
                 return Err(format!("Request failed with status code {}", res.status))

--- a/web/Cargo.toml
+++ b/web/Cargo.toml
@@ -16,13 +16,6 @@ path = "../"
 features = ["nightly"]
 
 [dependencies]
-iron = "0.4.0"
-router = "0.4.0"
-params = "0.4.0"
-handlebars = "0.20.0"
-handlebars-iron = "0.18.0"
-staticfile = "0.3.0"
-mount = "0.2.1"
 ipc-channel = "0.5.1"
 libc = "0.2.14"
 chrono = { version = "*", features = ["serde"] }
@@ -30,7 +23,15 @@ rustc-serialize = "*"
 serde_json = "0.8"
 serde_derive = "0.8"
 serde = "0.8.16"
-limiter = "0.2"
-logger = "0.1.0"
 url = "1.2.0"
 toml = { version = "0.2", features = ["serde"] }
+# iron crates
+iron = "0.5"
+router = "0.5"
+mount = "0.3"
+params = "0.6"
+handlebars = "0.25"
+handlebars-iron = "0.23"
+staticfile = "0.4"
+limiter = "0.3"
+logger = "0.3"

--- a/web/src/main.rs
+++ b/web/src/main.rs
@@ -21,7 +21,6 @@ extern crate limiter;
 extern crate logger;
 extern crate url;
 extern crate toml;
-#[macro_use]
 extern crate serde_derive;
 
 pub mod worker;
@@ -117,7 +116,6 @@ fn opensearch(rink: &Rink, _req: &mut Request) -> IronResult<Response> {
 }
 
 fn ifnot1helper(
-    c: &handlebars::Context,
     h: &handlebars::Helper,
     r: &Handlebars,
     rc: &mut handlebars::RenderContext
@@ -141,13 +139,12 @@ fn ifnot1helper(
         h.inverse()
     };
     match tmpl {
-        Some(ref t) => t.render(c, r, rc),
+        Some(ref t) => t.render(r, rc),
         None => Ok(()),
     }
 }
 
 fn urlescapehelper(
-    c: &handlebars::Context,
     h: &handlebars::Helper,
     r: &Handlebars,
     rc: &mut handlebars::RenderContext
@@ -157,15 +154,10 @@ fn urlescapehelper(
     use url::percent_encoding::{utf8_percent_encode, QUERY_ENCODE_SET};
 
     let tmpl = h.template();
-    let mut res = vec![];
-    match tmpl {
-        Some(ref t) => {
-            let mut new_rc = rc.with_writer(&mut res);
-            try!(t.render(c, r, &mut new_rc))
-        },
+    let res = match tmpl {
+        Some(ref t) => try!(t.renders(r, rc)),
         None => return Err(RenderError::new("urlescape is a block helper")),
-    }
-    let res = String::from_utf8_lossy(&res);
+    };
     let res = res.split_whitespace().collect::<Vec<_>>().join(" ");
     let res = utf8_percent_encode(&res, QUERY_ENCODE_SET).collect::<String>();
     let res = res.split("%20").collect::<Vec<_>>().join("+");
@@ -240,7 +232,9 @@ fn main() {
     let hbse = Arc::new(hbse);
     watch(&hbse);
 
-    let limiter = RequestLimit::new(5000, 5000);
+    let mut limiter = RequestLimit::default();
+    limiter.set_max_body_size(5000);
+    limiter.set_max_url_length(5000);
 
     chain.link_before(logger_before);
     chain.link_before(limiter);

